src/gatsby-node.js

```diff
@@ -213,9 +213,12 @@ exports.sourceNodes = async (
   reporter.info('Setting plugin status');
   reporter.info(JSON.stringify(newState));
 
-  if (websocketUpdates) {
+  const webSocketStart = function() {
     let ws = new WebSocket(baseUrl.replace(/(http)(s)?\:\/\//, 'ws$2://'));
     let timerId = null;
+    let count = 0;
+    let previousDelay = 1;
```

Comment:
instead of `let previousDelay = 1` the function should take this as an argument with default value (maybe `reconnectionDelay = 1`). Then it could pass it to reconnection function, which again would call this function with updated value.